Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp

```diff
@@ -202,14 +202,16 @@ RefPtr<DOMFormData> FetchBodyConsumer::packageFormData(ScriptExecutionContext* c
         CString boundary = boundaryWithDashes.utf8();
         size_t boundaryLength = boundary.length();
 
-        auto* currentBoundary = static_cast<const uint8_t*>(memmem(data.data(), data.size(), boundary.data(), boundaryLength));
-        if (!currentBoundary)
+        size_t currentBoundaryIndex = memmemSpan(data, boundary.span());
+        if (currentBoundaryIndex == notFound)
             return nullptr;
-        auto* nextBoundary = static_cast<const uint8_t*>(memmem(currentBoundary + boundaryLength, data.size() - (currentBoundary + boundaryLength - data.data()), boundary.data(), boundaryLength));
-        while (nextBoundary) {
-            parseMultipartPart(std::span { currentBoundary + boundaryLength, nextBoundary - currentBoundary - boundaryLength - strlen("\r\n") }, form.get());
-            currentBoundary = nextBoundary;
-            nextBoundary = static_cast<const uint8_t*>(memmem(nextBoundary + boundaryLength, data.size() - (nextBoundary + boundaryLength - data.data()), boundary.data(), boundaryLength));
+        data = data.subspan(currentBoundaryIndex + boundaryLength);
+        size_t nextBoundaryIndex = memmemSpan(data, boundary.span());
+        while (nextBoundaryIndex != notFound) {
+            parseMultipartPart(data.first(nextBoundaryIndex - strlen("\r\n")), form.get());
+            currentBoundaryIndex = nextBoundaryIndex;
+            data = data.subspan(nextBoundaryIndex + boundaryLength);
+            nextBoundaryIndex = memmemSpan(data, boundary.span());
         }
```

Comment:
Does something like this work, or is my syntax wrong?
```suggestion
        while (size_t nextBoundaryIndex = memmemSpan(data, boundary.span()); nextBoundaryIndex != notFound) {
            parseMultipartPart(data.first(nextBoundaryIndex - strlen("\r\n")), form.get());
            currentBoundaryIndex = nextBoundaryIndex;
            data = data.subspan(nextBoundaryIndex + boundaryLength);
        }
```
Even if we can’t quite do it this way, I think a version that doesn’t repeat the `memmemSpan` part of the loop is nicer. I guess maybe if the above doesn’t work it could be this:
```suggestion
        size_t nextBoundaryIndex;
        while ((nextBoundaryIndex = memmemSpan(data, boundary.span()) != notFound) {
            parseMultipartPart(data.first(nextBoundaryIndex - strlen("\r\n")), form.get());
            currentBoundaryIndex = nextBoundaryIndex;
            data = data.subspan(nextBoundaryIndex + boundaryLength);
        }
```

Reply:
Doesn't seem like a valid syntax